common/djangoapps/entitlements/api/v1/tests/test_views.py

```diff
@@ -0,0 +1,152 @@
+
+    def test_staff_user_required(self):
+        not_staff_user = UserFactory()
+        self.client.login(username=not_staff_user.username, password=UserFactory._DEFAULT_PASSWORD)
+        response = self.client.get(self.entitlements_list_url)
+        assert response.status_code == 403
+
+    def test_add_entitlement_with_missing_data(self):
+        entitlement_data_missing_parts = self._get_data_set(self.user, str(uuid.uuid4()))
+        entitlement_data_missing_parts.pop('mode')
+        entitlement_data_missing_parts.pop('course_uuid')
+
+        response = self.client.post(
+            self.entitlements_list_url,
+            data=json.dumps(entitlement_data_missing_parts),
+            content_type='application/json',
+        )
+        assert response.status_code == 400
+
+    def test_add_entitlement(self):
+        course_uuid = uuid.uuid4()
+        entitlement_data = self._get_data_set(self.user, str(course_uuid))
+
+        response = self.client.post(
+            self.entitlements_list_url,
+            data=json.dumps(entitlement_data),
+            content_type='application/json',
+        )
+        assert response.status_code == 201
+        results = response.data
+
+        course_entitlement = CourseEntitlement.objects.get(
+            user=self.user,
+            course_uuid=course_uuid
+        )
+
+        assert course_entitlement is not None
+        assert results['uuid'] == str(course_entitlement.uuid)
+        assert str(course_entitlement.course_uuid) == str(course_uuid)
+        assert course_entitlement.enrollment_course_run is None
+        assert course_entitlement.mode == 'verified'
+        assert course_entitlement.created is not None
+        assert course_entitlement.user.username == self.user.username
+
+    def test_get_entitlements(self):
+        uuid1 = CourseEntitlementFactory(user=self.user, order_number='TESTX-1001').uuid
+        uuid2 = CourseEntitlementFactory(user=self.user, order_number='TESTX-1002').uuid
+
+        response = self.client.get(
+            self.entitlements_list_url,
+            content_type='application/json',
+        )
+        assert response.status_code == 200
+
+        results = response.data.get('results', [])
+        assert results[0]['uuid'] == str(uuid1)
+        assert results[1]['uuid'] == str(uuid2)
+
+    def test_get_entitlement_by_uuid(self):
+
+        uuid1 = CourseEntitlementFactory(user=self.user, order_number='TESTX-1001').uuid
+        CourseEntitlementFactory(user=self.user, order_number='TESTX-1002')
+        url = reverse(self.entitlements_detail_path, args=[str(uuid1)])
+
+        response = self.client.get(
+            url,
+            content_type='application/json',
+        )
+        assert response.status_code == 200
+
+        results = response.data
+        assert results['uuid'] == str(uuid1)
+        serialized = CourseEntitlementSerializer(data=results)
+        assert serialized.is_valid()
```

Comment:
What you have here is basically validating the output as potential input to the serializer. You actually want to ensure the output matches the expected output.

`assert response.data == CourseEntitlementSerializer(course_entitlement)`